File: python/소수찾기.py
'''
https://programmers.co.kr/learn/courses/30/lessons/42839
완전탐색 소수찾기
'''
import itertools
import math

def primeNumber(num):
    if num <2:
        return False
    for i in range(2,int(math.sqrt(num))+1):
        if num%i==0:
            return False
    return True

def solution(numbers):
    #str을 리스트 요소로 분리하고
    numbers = [i for i in numbers]

    #그 리스트를 조합하고,
    numbers = [list(map(''.join,itertools.permutations(numbers,i))) for i in range(1,len(numbers)+1)]

    #조합한걸 중복을 없애고 (집합으로 변하고))
    numbersSet=set()
    for i in range(len(numbers)):
        for j in range(len(numbers[i])):
            numbersSet.add(int(numbers[i][j]))

    #집합을 다시 리스트로,
    numbers = list(numbersSet)

    #요소별로 검사
    answer = 0
    for number in numbers:
        if primeNumber(number):
            answer +=1
    return answer

File: nodejs/소수찾기.js
let answer = [];
const swap = (arr, i, n)=>{
    let temp = arr[n];
    arr[n] = arr[i];
    arr[i] = temp;
}
const getPrime = (arr, n, k)=>{

    if(n === arr.length-1){
        const str = arr.slice(0,k).join("");
        if(!answer.includes(str))answer.push(str);
        return;
    }else{
        for(let i=n; i<arr.length; i++){
            swap(arr, i, n);
            getPrime(arr, n+1, k); 
            swap(arr, i, n);
        }
        return answer;
    }
};
const isPrime = (n) =>{
   for(let i=2; i<=Math.sqrt(n); i++){
       if(n%i===0)return false;
   }
   return true;
}
function solution(numbers) {
    let count = 0;
    let answerArray = [];
    for(let i=0; i<numbers.length; i++){
       answerArray.push(getPrime(numbers.split(""), 0, i+1));
    }
    let answerZero = answerArray[0];
    let answerInt = answerZero
.map((value)=>Number(value))
.reduce((unique, item)=> unique.includes(item)? unique:[...unique, item],[]);
answerInt.forEach((value)=>{ if(isPrime(value) && value!==1 && value!==0)count++;});
    return count;
}